### PR TITLE
fix: fix KV implementations for payments and refunds

### PR DESCRIPTION
**Pull request by NishantJoshi00**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
This are minor changes to the overall implementation of the `RedisKV` to support the minimum required functionality. Bugs found and fixes includes:
- Wrong ordering of fields in `ReverseLookup` table
- Writing a workaround until `PaymentAttempt` queries support the single intent multiple attempt.
- Some keys received from reverse lookup not used while fetching from redis

### Additional Changes

- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Screenshot 2023-02-01 at 6 42 38 PM
Screenshot 2023-02-01 at 6 42 50 PM
Screenshot 2023-02-01 at 6 42 59 PM
Screenshot 2023-02-01 at 6 43 47 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/storage_models/src/reverse_lookup.rs [hunk 1]
```diff
@@ -13,10 +13,10 @@ use crate::schema::reverse_lookup;
 pub struct ReverseLookup {
     /// Primary key. The key id.
     pub lookup_id: String,
-    /// the value id. i.e the id you want to access KV table.
-    pub pk_id: String,
     /// the `field` in KV database. Which is used to differentiate between two same keys
     pub sk_id: String,
+    /// the value id. i.e the id you want to access KV table.
+    pub pk_id: String,
```
Comment (SanchithHegde) [inline crates/storage_models/src/reverse_lookup.rs:19]: Whoa, good catch. Thanks!

We'll need to seriously find reliable ways of catching such mistakes/differences with field ordering in struct definition against that in `schema.rs`.

**Review by SanchithHegde — APPROVED** [review]

File: crates/storage_models/src/reverse_lookup.rs (hunk 1 above)
Comment (NishantJoshi00) [reply crates/storage_models/src/reverse_lookup.rs:19]: Hey, not sure if this is helpful or not, but I was able to find

https://docs.rs/diesel-sort-struct-fields/latest/diesel_sort_struct_fields/

Maybe we can look into it....

**Review by Abhicodes-crypto — APPROVED** [review]
LGTM

**Review by dracarys18 — DISMISSED** [review]

File: crates/router/src/db/payment_attempt.rs [hunk 2]
```diff
@@ -545,6 +556,10 @@ mod storage {
                 }
 
                 enums::MerchantStorageScheme::RedisKv => {
+                    /*
+                     * BREAKING CHANGE:
```
Comment (NishantJoshi00) [inline crates/router/src/db/payment_attempt.rs:560]: @Abhicodes-crypto, once this function supports multiple attempt do the necessary changes instructed in the BREAKING_CHANGE comments

**Review by dracarys18 — APPROVED** [review]

**Review by Abhicodes-crypto — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by dracarys18 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**